lib/supers/destination.js

```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+const _ = require('lodash');
+/**
+ * Destination plugins control how and where we write our transformed property.
+ * Default case is to write to an object property in the scheme in play
+ * and then to
+ * push the scheme in play to the newScheme place.
+ *
+ * Another behavior would be to write to a file. <- this could be useful in
+ *  situations where we are creating cached destination writing output.
+ * Another behavior would be to use a destination token template to write
+ * the transformed property into a tokenized template
+ * essentially adding that shape if it doesn't exist to the scheme in play
+ * and if it does exist adding the property to the nested location within.
+ *
+ * Lets get write to object property down
+ * Then merge property
+ * then aim for tokenized templates
+ * then write to a file.
+ *
+ * We may need to bring in the schemePunk creation because maybe that's
+ * actually some behavior of destination or that destination enhances that
+ * in some way.
+ */
+module.exports = class destinationSuperBase {
+  /**
+   * Initialize the destination super class.
+   *
+   * @param {Object} options - An options object.
+   * @param {*} transformedValue - A transformed target.
+   * @param {Object} scheme - A destination scheme.
+   * @param {Object} holdOvers - An object of holdovers.
+   */
+  init(options, transformedValue, scheme, holdOvers) {
+    // eslint-disable-next-line no-unused-vars
+    this.scheme = scheme;
+    this.setTarget(options.target);
+    this.setDestination(options.destinationValue);
+    this.setValue(transformedValue);
+    this.setHoldOvers(holdOvers);
```

Comment:
If this should be async, you should return a `Promise.resolve(this)` here, allowing for a pattern like this: 

```javascript
new Thing().init().then(inst => inst.doNextThing())
```

Reply:
done for each of the classes source, transform, and destination.